# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationUtils.h
# Commit: 8fda62a34fb2
# Full Hash: 8fda62a34fb24ee9af0cc61f0d1584f35e64a083
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2025-07-17 09:30:54
# Regressor Bug: 1972117
# File Overlap Count: 1
# Description:
#   Bug 1972117 - Part 3: Move NotificationsCallback to NotificationUtils r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D255860
# ==============================================================================

diff -r f2e7fc3530d1 -r 8fda62a34fb2 dom/notification/NotificationUtils.h
--- a/dom/notification/NotificationUtils.h	Wed Jul 16 19:52:22 2025 +0000
+++ b/dom/notification/NotificationUtils.h	Wed Jul 16 19:52:23 2025 +0000
@@ -69,6 +69,12 @@
 
 nsCOMPtr<nsINotificationStorage> GetNotificationStorage(bool isPrivate);
 
+using NotificationsPromise =
+    MozPromise<CopyableTArray<IPCNotification>, nsresult, false>;
+
+already_AddRefed<NotificationsPromise> GetStoredNotificationsForScope(
+    nsIPrincipal* aPrincipal, const nsACString& aScope, const nsAString& aTag);
+
 nsresult GetOrigin(nsIPrincipal* aPrincipal, nsString& aOrigin);
 
 nsresult PersistNotification(nsIPrincipal* aPrincipal,